logs/querysample/normalize.go

```diff
@@ -0,0 +1,242 @@
+	StartupCost                 *float64                 `json:"Startup Cost,omitempty"`
+	Strategy                    *string                  `json:"Strategy,omitempty"`     // does not need normalize (fixed set of possible values)
+	SubplanName                 *string                  `json:"Subplan Name,omitempty"` // does not need normalize (generated name by planner)
+	SubplansRemoved             *int64                   `json:"Subplans Removed,omitempty"`
+	TableFunctionCall           *string                  `json:"Table Function Call,omitempty"` // needs normalize (value comes from show_expression)
+	TableFunctionName           *string                  `json:"Table Function Name,omitempty"` // does not need normalize (name of a function)
+	TempReadBlocks              *int64                   `json:"Temp Read Blocks,omitempty"`
+	TempWrittenBlocks           *int64                   `json:"Temp Written Blocks,omitempty"`
+	TIDCond                     *string                  `json:"TID Cond,omitempty"`        // needs normalize (value comes from show_scan_qual)
+	TuplestoreName              *string                  `json:"Tuplestore Name,omitempty"` // does not need normalize (name of an emphemeral named relation, aka ENR) -- note that there is no test exercising this, since its not clear how this part of explain.c can get called
+	TotalCost                   *float64                 `json:"Total Cost,omitempty"`
+	TuplesInserted              *float64                 `json:"Tuples Inserted,omitempty"`
+	WALBytes                    *uint64                  `json:"WAL Bytes,omitempty"` // Note this is the only *unsigned* integer field currently
+	WALFPI                      *int64                   `json:"WAL FPI,omitempty"`
+	WALRecords                  *int64                   `json:"WAL Records,omitempty"`
+	WorkerNumber                *int64                   `json:"Worker Number,omitempty"`
+	Workers                     *[]planNode              `json:"Workers,omitempty"`
+	WorkersLaunched             *int64                   `json:"Workers Launched,omitempty"`
+	WorkersPlanned              *int64                   `json:"Workers Planned,omitempty"`
+}
+
+func normalizeExpr(expr *string) {
+	if expr == nil {
+		return
+	}
+	res := util.NormalizeQuery("SELECT "+*expr, "unparsable", -1)
+	if res == util.QueryTextUnparsable {
+		*expr = util.QueryTextUnparsable
+		return
+	}
+
+	*expr = strings.TrimPrefix(res, "SELECT ")
+}
+
+func normalizeSortKey(expr *string) {
+	if expr == nil {
+		return
+	}
+	res := util.NormalizeQuery("SELECT ORDER BY "+*expr, "unparsable", -1)
+	if res == util.QueryTextUnparsable {
+		*expr = util.QueryTextUnparsable
+		return
+	}
+
+	*expr = strings.TrimPrefix(res, "SELECT ORDER BY ")
+}
+
+func normalizeExprArray(exprArray *[]string) {
+	if exprArray == nil {
+		return
+	}
+	for idx, expr := range *exprArray {
+		normalizeExpr(&expr)
+		(*exprArray)[idx] = expr
```

Comment:
Is this still necessary? You're updating the expr by reference, no? But NBD.

Reply:
Yeah, this is one of the Go-isms that I don't like..

This is necessary (and in fact a test fails if this line is not present), because `&expr` references the location of the `expr` variable, which is a copy of the list element - not the list element itself. Essentially `normalizeExpr` just does `expr = ...` but that won't modify the list itself (since expr is not a pointer, but a simple string).